File: src/test/kotlin/study/old/Person.java
package study.old;

public class Person {
    private final String name;
    private final int age;
    private String nickname;

    public Person(final String name, final int age, final String nickname) {
        this.name = name;
        this.age = age;
        this.nickname = nickname;
    }

    public String getName() {
        return name;
    }

    public int getAge() {
        return age;
    }

    public String getNickname() {
        return nickname;
    }

    public void setNickname(final String nickname) {
        this.nickname = nickname;
    }
}

File: src/main/kotlin/study/Person.kt
package study

// 코틀린은 프로퍼티 기반의 언어고 프로퍼티는 getter와 setter가 관행적으로 붙는다.
data class Person(val name: String, val age: Int, var nickname: String? = "김수현") // 필드가 아닌 프로퍼티
